sqlite/ext/comdb2/table_properties.c

```diff
@@ -0,0 +1,87 @@
+/*
+   Copyright 2021 Bloomberg Finance L.P.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+#include <stdlib.h>
+#include <string.h>
+#include <stddef.h>
+#include "comdb2.h"
+#include "sqlite3.h"
+#include "ezsystables.h"
+#include "schema_lk.h"
+
+extern struct dbenv *thedb;
+
+sqlite3_module systblTablePropertiesModule = {
+    .access_flag = CDB2_ALLOW_USER,
+};
+
+typedef struct systable_table_properties {
+    char *table_name;
+    const char *odh;
+    const char *compress;
+    const char *blob_compress;
+    const char *in_place_updates;
+    const char *instant_schema_change;
+} systable_table_properties_t;
+
+int table_properties_systable_collect(void **data, int *nrecords)
+{
+    rdlock_schema_lk();
```

Comment:
You can't simply grab schema_lk here. The thread might have already held berkdb locks and thus may deadlock with a schema change. Consider using "comdb2_systable" table lock instead (which is a berkdb lock, too). See [this example](https://github.com/bloomberg/comdb2/blob/master/sqlite/ext/comdb2/columns.c#L320).